runtime/src/append_vec.rs

```diff
@@ -31,6 +33,8 @@ pub struct StoredMeta {
     pub data_len: u64,
 }
 
+/// This struct will be backed by mmaped and snapshotted data files.
+/// So the data layout must be stable and consistent across the entire cluster!
```

Comment:
Rather than comments, I wonder if we should define an attribute to start clearly structs that are part of the Solana ABI, like `#[solana_abi]`.   One day this'll then allow us to programmatically find all these structs and compare their values between release to flag ABI changes.  

This is totally out of scope of this PR but if that sounds good, @ryoqun do you want to think about this more and make a [design proposal](https://github.com/solana-labs/solana/blob/master/CONTRIBUTING.md#design-proposals) for how we might start to track ABI changes

Reply:
Heh, let's talk about ABI compat! That task looks interesting to me! I'll give it a try!

This reminds of my rookie days when I found an ABI-related bug: https://github.com/solana-labs/solana/pull/6388 Always, these are pesky bugs.. Hopefully, I can find good solution!